libsolidity/formal/CHC.cpp

```diff
@@ -654,6 +688,91 @@ vector<VariableDeclaration const*> CHC::stateVariablesIncludingInheritedAndPriva
 	return stateVars;
 }
 
+void CHC::createArrayLengths(Type const& _type)
+{
+	set<Type const*> types;
+	queue<Type const*> toVisit;
+	set<Type const*> visited;
+	toVisit.push(&_type);
+	while (!toVisit.empty())
```

Comment:
Maybe use algorithms::BFS here

Reply:
Yep, looks like a case for the algorithm.

Reply:
Yea, I was just waiting for you to mention it to check how to use it :p